Index: src/App.jsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/App.jsx b/src/App.jsx
--- a/src/App.jsx	
+++ b/src/App.jsx	(date 1669265923494)
@@ -1,19 +1,24 @@
-import React, { lazy, Suspense, useEffect, useState } from "react";
-import { Route, Switch, useLocation } from "react-router-dom";
+import React, { Suspense, useEffect, useState } from "react";
+import { Redirect, Route, Switch, useHistory, useLocation } from "react-router-dom";
 import DefaultLayout from "@components/layout/DefaultLayout";
 import Login from "@pages/login";
 import Spinner from "@components/Spinner";
+import { lazy } from "@loadable/component";
+
 
 import { ErrorBoundary } from "react-error-boundary";
 import Error from "@components/error/Error";
 import { GET_ROUTES } from "@pages/MODULE/SYS/ROLE/ROLE0030/api/menu";
 import { getCookie } from "@utils/cookieUtil";
+import Main from "@pages/main";
 
 
 const App = () => {
   const [routes, setRoutes] = useState([]);
+  const history = useHistory();
   const location = useLocation();
 
+
   // if (!getCookie("_at")) {
   //   return <Login />;
   // }
@@ -31,23 +36,18 @@
     <Switch>
       <Route path="/login" render={() => <Login />} />
       <Suspense fallback={<Spinner />}>
-        {/* <ErrorBoundary fallbackRender={Error}> */}
+        {/*  <ErrorBoundary fallbackRender={Error}> */}
         <DefaultLayout>
+          <Route path="/" exact render={() => <Main />} />
           <Switch location={location}>
             {routes.map(({ menuNo, menuPath }) => {
               return (
                 <Route
                   exact
                   path={menuPath}
-                  component={lazy(() => import(`@pages/MODULE${menuPath}`).catch(() => {
-                    
-                  }))}
-                  // (err) => {
-                  //   console.log(err);
-                  //   console.log("good");
-                  // },
-
-                  // key={menuNo}
+                  component={lazy(() => import(`@pages/MODULE${menuPath}`)
+                    .catch(() => { return history.push("/"); }))}
+                  key={menuNo}
                 />
               );
             })}
